Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3079,12 +3080,19 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
             else
                 [allJSONData addObject:dynamicRulesAsData];
 
+            HashSet<double> dynamicRuleIDs;
+            for (NSDictionary<NSString *, id> *rule in rules)
+                dynamicRuleIDs.add(objectForKey<NSNumber>(rule, @"id").doubleValue);
+
+            m_dynamicRulesIDs = dynamicRuleIDs;
```

Comment:
You don't need `dynamicRuleIDs`, you can use `m_dynamicRulesIDs` directly in the loop.

Reply:
We are doing this to replace the whole set, so we don't have the track the removes.